### PR TITLE
core: header cleanup

### PR DESCRIPTION
**Pull request by kaspar030**
```kernel.h``` was long overdue for removal. This PR removes it (and shuffles some declarations into the correct files).

### CODE REVIEW

File: cpu/native/native_cpu.c [hunk 1]
```diff
@@ -59,6 +57,9 @@
 extern netdev2_tap_t netdev2_tap;
 #endif
 
+/* cannot include sched.h due to system header name clash */
+extern NORETURN void sched_task_exit(void);
```
Comment (OlegHahm) [inline cpu/native/native_cpu.c]: How unfortunate.

**Comment by OlegHahm** [comment]
ACK if Travis is fine.

File: cpu/native/native_cpu.c (hunk 1 above)
Comment (kaspar030) [inline cpu/native/native_cpu.c]: and actually wrong ;)

**Comment by kaspar030** [comment]
Travis is fine. Would someone take a second look?

**Comment by gebart** [comment]
Changes look OK. You have my ACK
Travis is OK as well.

**Comment by kaspar030** [comment]
Nice, thanks for reviewing!

**Pull request merged**